cmd/subnetcmd/upgradecmd/vm_test.go

```diff
@@ -274,3 +282,53 @@ func TestAtMostOneAutomationSelected(t *testing.T) {
 		})
 	}
 }
+
+func TestUpdateToCustomBin(t *testing.T) {
+	assert := require.New(t)
+	testDir := t.TempDir()
+
+	subnetName := "testSubnet"
+	sc := models.Sidecar{
+		Name:       subnetName,
+		VM:         models.SubnetEvm,
+		VMVersion:  "v3.0.0",
+		RPCVersion: 20,
+		Subnet:     subnetName,
+	}
+	networkToUpgrade := futureDeployment
+
+	factory := logging.NewFactory(logging.Config{})
+	log, err := factory.Make("avalanche")
+	assert.NoError(err)
+
+	// create the user facing logger as a global var
+	ux.NewUserLog(log, os.Stdout)
+
+	app = &application.Avalanche{}
+	app.Setup(testDir, log, config.New(), prompts.NewPrompter(), application.NewDownloader())
```

Comment:
Note: not for this PR, but we need to repeat this test setup code all over always. Going to create a ticket to create a testutil helper

Reply:
https://github.com/ava-labs/avalanche-cli/issues/565

Reply:
There is an `internal/testutils` package where I started to do this